src/modules/ctc/src/ctc_callback_handler.cc

```diff
@@ -0,0 +1,181 @@
+        block_entry->push_back(slint::StandardListViewItem(std::to_string(block.block).c_str()));
+        block_entry->push_back(slint::StandardListViewItem(block.maintenance ? "Maintenance" : "Open"));
+        block_entry->push_back(slint::StandardListViewItem("_"));
+        block_entry->push_back(slint::StandardListViewItem(block.occupied ? "Occupied" : "_"));
+        block_entry->push_back(slint::StandardListViewItem(block.power_failure ? "Failure" : "_"));
+        block_data_model->push_back(block_entry);
+    }
+    ctc_ui->set_block_data(block_data_model);
+    // Create Model to Populate Train Schedule Table
+    auto train_schedule_model = std::make_shared<slint::VectorModel<std::shared_ptr<slint::Model<slint::StandardListViewItem>>>>();
+    ctc_ui->set_train_schedules(train_schedule_model);
+
+    // Create Model to Populate Trains Dropdown
+    auto train_ids = std::make_shared<slint::VectorModel<slint::SharedString>>();
+    train_ids->push_back(slint::SharedString("New Train"));
+    ctc_ui->set_trains(train_ids);
+
+    // Create Model to Populate Stations
+    auto                      stations_model = std::make_shared<slint::VectorModel<slint::SharedString>>();
+    std::vector<ctc::Station> stations       = ctc.GetStations();
+    for (const ctc::Station &station : stations)
+    {
+        stations_model->push_back(station.station_name.c_str());
+    }
+    ctc_ui->set_stations(stations_model);
+}
+
+// This will be called in the backend each time through the loop
+void backend_handler(ctc::Ctc &ctc_office, slint::ComponentHandle<ui::CtcUi> &ctc_ui)
+{
+    manual_dispatch_handler(ctc_office, ctc_ui);
+
+    // TODO repeat for each callback that needs to be handled in the backend
+}
+
+static void register_callbacks(slint::ComponentHandle<ui::CtcUi> &ctc_ui)
+{
+    ctc_ui->on_manual_dispatch([&ctc_ui] {
+            manual_dispatch_callback(ctc_ui);
+        });
+
+    // TODO repeat for each event listener
+}
+
+// Callbacks run on the frontend thread
+static inline void manual_dispatch_callback(slint::ComponentHandle<ui::CtcUi> &ctc_ui)
+{
+    destination_channel.Send(std::string(ctc_ui->get_destination()));
+    train_id_channel.Send(std::string(ctc_ui->get_train_id()));
+
+    // TODO update UI if necessary
+
+}
+
+// Handlers run on the backend thread
+static void manual_dispatch_handler(ctc::Ctc &ctc_office, slint::ComponentHandle<ui::CtcUi> &ctc_ui)
+{
+    if (destination_channel.DataAvailable())
+    {
+        // TODO need a function that instantiates a new train model and train controller with the same id
+        std::string train_id    = train_id_channel.Receive();
+        std::string destination = destination_channel.Receive();
+        std::string current_position;
+        std::string authority;
+        std::string suggested_speed;
+
+        // Dispatch new train
+        if (train_id == "New Train")
+        {
+            // Backend Dispatch
+            ctc::Train new_train;
```

Comment:
We should have some sort of builder function to instantiate a new train model and train controller so we can assign the same ID to all objects associated with a particular train.  Also what is the purpose of `new_train`?  Seems like all we need is a new train ID?

Reply:
For now, `ctc::Train` struct default constructor create a train with new ID based on the previously created train. If there're 2 train created, next new train will have ID of 3. I agree that we should have some builder function to integrate new train model and train controller. Thinking I'd do that in a seperate PR